src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,555 @@
+    uint256 private constant CREATOR_OFFSET_MULTIPLIER = uint256(2)**(256 - 160);
+    uint256 private constant SUBID_MULTIPLIER = uint256(2)**(256 - 224);
+    // ((uint256(1) * 2**224) - 1) << 32;
+    uint256 private constant BASEID_MASK = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF00000000;
+    // ((uint256(1) * 2**32) - 1) << 200;
+    uint256 private constant VERSION_MASK = 0x000000FFFFFFFF00000000000000000000000000000000000000000000000000;
+    bytes32 private constant base32Alphabet = 0x6162636465666768696A6B6C6D6E6F707172737475767778797A323334353637;
+
+    mapping(uint256 => mapping(uint256 => uint256)) private _gameAssets;
+    mapping(address => address) private _creatorship; // creatorship transfer
+
+    mapping(uint256 => bytes32) private _metaData;
+    mapping(address => mapping(address => bool)) private _gameEditors;
+
+    ///////////////////////////////  Events //////////////////////////////
+
+    /// @dev Emits when a game is updated.
+    /// @param oldId The id of the previous erc721 GAME token.
+    /// @param newId The id of the newly minted token.
+    /// @param update The changes made to the Game: new assets, removed assets, uri
+
+    event GameTokenUpdated(uint256 indexed oldId, uint256 indexed newId, IGameToken.Update update);
+
+    /// @dev Emits when creatorship of a GAME token is transferred.
+    /// @param original The original creator of the GAME token.
+    /// @param from The current 'creator' of the token.
+    /// @param to The new 'creator' of the token.
+    event CreatorshipTransfer(address indexed original, address indexed from, address indexed to);
+
+    /// @dev Emits when an address has its gameEditor status changed.
+    /// @param gameOwner The owner of the GAME token.
+    /// @param gameEditor The address whose editor rights to update.
+    /// @param isEditor WHether the address 'gameEditor' should be an editor.
+    event GameEditorSet(address indexed gameOwner, address gameEditor, bool isEditor);
+
+    constructor(
+        address metaTransactionContract,
+        address admin,
+        IAssetToken asset,
+        address initialMinter
+    ) ERC721BaseToken(metaTransactionContract, admin) {
+        _asset = asset;
+        _minter = initialMinter;
+    }
+
+    ///////////////////////////////  Modifiers //////////////////////////////
+
+    modifier notToZero(address to) {
+        require(to != address(0), "DESTINATION_ZERO_ADDRESS");
+        _;
+    }
+
+    modifier notToThis(address to) {
+        require(to != address(this), "DESTINATION_GAME_CONTRACT");
+        _;
+    }
+
+    ///////////////////////////////  Functions //////////////////////////////
+
+    /// @notice Allow token owner to set game editors.
+    /// @param gameCreator The address of a GAME token creator.
+    /// @param editor The address of the editor to set.
+    /// @param isEditor Add or remove the ability to edit.
+    function setGameEditor(
+        address gameCreator,
```

Comment:
minor but owner or gameOnwer would be more accurate

Reply:
this is from me :)